eskip/predicates.go

```diff
@@ -49,13 +49,24 @@ func SinglePredicateByName(p []*Predicate, name string) (*Predicate, error) {
 
 // ValidatePredicates returns an error when certain predicates are added multiple times.
 func ValidatePredicates(pp []*Predicate) error {
+	counts := make(map[string]int)
 	for _, p := range pp {
-		switch p.Name {
-		case "Path", "Weight":
-			if len(AllPredicatesByName(pp, p.Name)) > 1 {
-				return fmt.Errorf("predicate of type %s can only be added once", p.Name)
-			}
+		if _, ok := counts[p.Name]; !ok {
```

Comment:
this is unnecessary in go because the map returns the defaul value when the key doesn't exist. This is enough:

```
counts[p.Name]++
```

Reply:
Awesome, TIL :)